src/sagas.js

```diff
@@ -344,6 +344,34 @@ function* refreshData (action) {
 }
 
 function* fetchVmsByPage (action) {
+  const actual = Selectors.getOvirtVersion().toJS()
+  if (compareVersion({ major: parseInt(actual.major), minor: parseInt(actual.minor) }, { major: 4, minor: 2 })) {
+    yield fetchVmsByPageV42(action)
+  } else {
+    yield fetchVmsByPageLower(action)
+  }
+}
+
+function* fetchVmsByPageV42 (action) {
+  const { shallowFetch, page } = action.payload
+  let additional = []
+  if (!shallowFetch) {
+    additional = ['cdroms', 'sessions', 'disk_attachments.disk', 'graphics_consoles']
+  }
+  action.payload.additional = additional
+  // TODO: paging: split this call to a loop per up to 25 vms
+  const allVms = yield callExternalAction('getVmsByPage', Api.getVmsByPage, action)
+
+  if (allVms && allVms['vm']) { // array
+    const internalVms = allVms.vm.map(vm => Api.vmToInternal({ vm, getSubResources: true }))
+
+    yield put(updateVms({ vms: internalVms, copySubResources: true, page: page }))
+  }
+
+  yield put(persistState())
+}
+
+function* fetchVmsByPageLower (action) {
```

Comment:
We should use generally understandable naming convention.
I propose always start with capital `V` followed by version to avoid ambiguity in different context.

Means, this function shall be fetchVmsByPageVLower